src/foam/dao/GUIDDAO.js

```diff
@@ -69,8 +77,7 @@ return (foam.core.PropertyInfo)(getOf().getAxiomByName(getProperty()));
 Object val = obj.getProperty(getProperty());
 
 if ( "".equals(val) ) {
-  Random r = ThreadLocalRandom.current();
-  getAxiom().set(obj, new UUID(r.nextLong(), r.nextLong()).toString());
+  getAxiom().set(obj, getUIDGenerator().generate());
```

Comment:
might need the same UID generator for JS.

Reply:
Only if we had a Node.js server, otherwise, we couldn't rely on different clients to not produce the same IDs.

Reply:
Also, I would leave GUIDDAO alone, and make a new FUIDAO.